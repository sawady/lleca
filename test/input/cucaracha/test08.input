

fun g(x : Int) {
  putChar(x)
  putChar(x)
  putChar(x)
}

fun f(x : Int, y : Int) {
  g(x)
  g(y)
  g(x)
  putChar(10)
}

fun main() {
  f(65, 66)
  f(66, 67)
  f(67, 68)
}
