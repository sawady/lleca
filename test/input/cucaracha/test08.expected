Cons(
  Function(
    g,
    Cons(
      Parameter(
        x,
        Int
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtCall(
        putChar,
        Cons(
          ExprVar(
            x
          ),
          Nil
        )
      ),
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprVar(
              x
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprVar(
                x
              ),
              Nil
            )
          ),
          Nil
        )
      )
    )
  ),
  Cons(
    Function(
      f,
      Cons(
        Parameter(
          x,
          Int
        ),
        Cons(
          Parameter(
            y,
            Int
          ),
          Nil
        )
      ),
      Unit,
      Cons(
        StmtCall(
          g,
          Cons(
            ExprVar(
              x
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            g,
            Cons(
              ExprVar(
                y
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              g,
              Cons(
                ExprVar(
                  x
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    10
                  ),
                  Nil
                )
              ),
              Nil
            )
          )
        )
      )
    ),
    Cons(
      Function(
        main,
        Nil,
        Unit,
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstNum(
                65
              ),
              Cons(
                ExprConstNum(
                  66
                ),
                Nil
              )
            )
          ),
          Cons(
            StmtCall(
              f,
              Cons(
                ExprConstNum(
                  66
                ),
                Cons(
                  ExprConstNum(
                    67
                  ),
                  Nil
                )
              )
            ),
            Cons(
              StmtCall(
                f,
                Cons(
                  ExprConstNum(
                    67
                  ),
                  Cons(
                    ExprConstNum(
                      68
                    ),
                    Nil
                  )
                )
              ),
              Nil
            )
          )
        )
      ),
      Nil
    )
  )
)
